spec/instance_title_fields_spec.rb

```diff
@@ -87,6 +88,119 @@
+        expect(doc_hash[:title_245_search]).to be_a String
+        expect(doc_hash[:title_245_search]).to eq 'foo'
+      end
+      it 'takes first label if there are multiple values' do
+        solutions << RDF::Query::Solution.new(p: 'rdf-schema#label', o: 'foo')
+        solutions << RDF::Query::Solution.new(p: 'rdf-schema#label', o: 'bar')
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_245_search]).to eq 'foo'
+      end
+      it 'nil if there is no label' do
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_245_search]).to eq nil
+      end
+    end
+
+    context 'title_full_display' do
+      let(:sparql_conn) { double('sparql client') }
+      before(:each) do
+        allow(isd).to receive(:sparql).and_return(sparql_conn)
+      end
+      it 'is a resposibilityStatement String (not an Array - single valued Solr field)' do
+        solutions << RDF::Query::Solution.new(p: 'responsibilityStatement', o: 'foo')
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_full_display]).to be_a String
+        expect(doc_hash[:title_full_display]).to eq ' / foo'
+      end
+      it 'takes first resposibilityStatement if there are multiple values' do
+        solutions << RDF::Query::Solution.new(p: 'responsibilityStatement', o: 'foo')
+        solutions << RDF::Query::Solution.new(p: 'responsibilityStatement', o: 'bar')
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_full_display]).to eq ' / foo'
+      end
+      it 'strips "/" if there is no responsibilityStatement' do
+        solutions << RDF::Query::Solution.new(p: 'mainTitle', o: 'foo')
+        solutions << RDF::Query::Solution.new(p: 'subtitle', o: 'bar')
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        expect(doc_hash[:title_full_display]).to eq 'foo : bar'
+      end
+      it 'removes trailing punctuation' do
+        ['\\', ',', ':', ';', '/'].each do |punct|
+          solutions = RDF::Query::Solutions.new
+          solutions << RDF::Query::Solution.new(p: 'responsibilityStatement', o: "foo#{punct}")
+          expect(sparql_conn).to receive(:query).and_return(solutions)
+          allow(isd).to receive(:sparql).and_return(sparql_conn)
+          doc_hash = isd.send(:add_doc_title_fields, {})
+          expect(doc_hash[:title_full_display]).to eq ' / foo'
+        end
+        solutions = RDF::Query::Solutions.new
+        solutions << RDF::Query::Solution.new(p: 'responsibilityStatement', o: "foo : ")
+        expect(sparql_conn).to receive(:query).and_return(solutions)
+        allow(isd).to receive(:sparql).and_return(sparql_conn)
+        doc_hash = isd.send(:add_doc_title_fields, {})
+        expect(doc_hash[:title_full_display]).to eq ' / foo'
+      end
+    end
+
+    context 'title_display' do
```

Comment:
similar to above.  In fact, I would put the title_display tests before title_full_display tests.  partly because they are that order in the code, and partly because title_full_display depends on title_display